packages/authorization-process/test/addClientPurpose.test.ts

```diff
@@ -0,0 +1,557 @@
+/* eslint-disable @typescript-eslint/no-floating-promises */
+import {
+  decodeProtobufPayload,
+  getMockAgreement,
+  getMockClient,
+  getMockDescriptor,
+  getMockDocument,
+  getMockEService,
+  getMockPurpose,
+  getMockPurposeVersion,
+  getMockTenant,
+  writeInReadmodel,
+} from "pagopa-interop-commons-test";
+import { describe, expect, it } from "vitest";
+import {
+  Agreement,
+  Client,
+  ClientPurposeAddedV2,
+  Descriptor,
+  Purpose,
+  agreementState,
+  descriptorState,
+  generateId,
+  purposeVersionState,
+  toClientV2,
+  toReadModelAgreement,
+  toReadModelEService,
+  toReadModelPurpose,
+} from "pagopa-interop-models";
+import { genericLogger } from "pagopa-interop-commons";
+import {
+  agreementNotFound,
+  clientNotFound,
+  eserviceNotFound,
+  noVersionsFoundInPurpose,
+  organizationNotAllowedOnClient,
+  organizationNotAllowedOnPurpose,
+  purposeAlreadyLinkedToClient,
+  purposeNotFound,
+} from "../src/model/domain/errors.js";
+import {
+  addOneClient,
+  agreements,
+  authorizationService,
+  eservices,
+  purposes,
+  readLastAuthorizationEvent,
+} from "./utils.js";
+
+describe("addClientPurpose", async () => {
+  it("should write on event-store for the addition of a purpose into a client", async () => {
+    const mockDescriptor: Descriptor = {
+      ...getMockDescriptor(),
+      state: descriptorState.published,
+      interface: getMockDocument(),
+      publishedAt: new Date(),
+    };
+
+    const mockEservice = {
+      ...getMockEService(),
+      descriptors: [mockDescriptor],
+    };
+    const mockConsumer = getMockTenant();
```

Comment:
```suggestion
    const consumerId = generateId();
```
We only need the `id`; same for the other tests

Reply:
I think it's clearer to create the entire object

Reply:
Updated